datarobot_batch_scoring/utils.py

```diff
@@ -410,7 +413,8 @@ def investigate_encoding_and_dialect(dataset, sep, ui, fast=False,
     csv.register_dialect('dataset_dialect', dialect)
     #  the csv writer should use the systems newline char
     csv.register_dialect('writer_dialect', dialect,
-                         lineterminator=os.linesep)
+                         lineterminator=os.linesep,
+                         delimiter=(output_delimiter or dialect.delimiter))
```

Comment:
I think validating that the output_delimiter is a bytestring in Python2 is a necessary change.